### ISSUE TITLE
Improve Trigger in rule to better deal with DataType

### ISSUE THREAD

**Issue by cfabianski** [Bearer/bearer#744]
We currently support 4 different types for `trigger` when writing a rule (`presence`, `absence`, `local` and `global`).

**current**

```yaml
# rule.yml
trigger: local | global | presence | absence
```

## History

We started with only 2 triggers `global` and `local`. So we were firing the rule when we had the patterns matching and we detected some data types either globally or locally.
We then introduced the new trigger `presence` to fire no matter if we had detected some data types or not.

The `presence` is very close to the `global` trigger (especially since we refactored the severity calculation), expect that a `presence` rule can fire even when there are no data types detected globally.

Later on, we introduced the `absence` trigger. This is when you have an auxiliary pattern that's matching but the whole global pattern doesn't. Again, that's not taking into account the data types.

In the end, it feels like we are mixing up 2 things:
- when do we trigger? (`absence` or `presence`)
- should we care about the data types at all? if so, locally or globally?

## Proposal

We could make it a bit easier to read by making the following changes

**proposal**

```yaml
trigger:
  pattern: present | absent # or missing instead of `absent`
  pattern_initializer: <name-of-auxiliary-rule> # would be used in conjunction with `absent` to indicate what is the minimal setup to fire the rule in case the pattern is missing. Replaces `trigger_rule_on_presence_of`
  data_types: global # or not present
```

The `local` use case could be deducted from the rule itself. If we have in the filters `data_types` as a detection we would consider the local scope
A missing `data_types` node, would indicate that we wouldn't take the data types into account to fire the rule. 

## References

- https://github.com/Bearer/bearer/pull/727

**Comment by gmontard** [comment]
I'm not a huge fan of the "present", "absent" naming for the ```pattern``` values.

Since the pattern is somehow a regexp (_that's how we all think about it_), this is here ultimately all about the fact it matches or doesn't match, so why not use this term that every developer I believe understand in a regexp context? 

```
pattern: match | not_match
```
_Or in plain English: Does this pattern match?_ 

Though for, ```data_types```, I would prefer to use the naming  "present" that sounds easier to understand in this context:
```
data_types: present # or n/a or even absent if you want.
```

_Or in plain English: Are date_types present or not?_

Good naming is all about the fact that people able to understand what it does without knowing how the internal works.

**Comment by elsapet** [comment]
I have some alternative wording suggestions for consideration.

I'm in favour of the `presence` / `absence` terminology for pattern-matching because I think there may be a difference between a pattern not matching (this would be the case for many lines in the code base) and the absence throughout the code base of any match (which is what we're after), but I agree `pattern: presence` isn't as self-explanatory as it could be. 

What do we think about the following alternative: 

```
trigger: 
 - match_on: presence | absence # default to presence
 - required_detection: some_aux_rule_id # used for absence matches only; default to nil
```

For the `data_types`, is there a use case for `data_types: absent` that we have in mind? 

if not, I'd propose something like this

```
trigger:
 - data_types_required: true | false # default to false
```

For me, the question is _are data_types required or not?_ If they're not required, we don't care either way if they are present or absent. I think this also keeps us away from having to handle nonsensical situations where we have `data_types: absent` in the trigger, but `$<DATA_TYPE>` appearing in the patterns. 

Looking forward to your thoughts @gmontard @cfabianski 🙏

**Comment by gmontard** [comment]
I like it @elsapet 💯
